src/views/about.nim

```diff
@@ -1,22 +1,21 @@
+import os
 import karax/[karaxdsl, vdom]
 import markdown
 
+import ".."/[types]
+
 const
   hash = staticExec("git log -1 --format=\"%h\"")
   link = "https://github.com/zedeus/nitter/commit/" & hash
 
-let
-  about = markdown(readFile("public/md/about.md"))
-  feature = markdown(readFile("public/md/feature.md"))
-
-proc renderAbout*(): VNode =
+proc renderAbout*(cfg: Config): VNode =
   buildHtml(tdiv(class="overlay-panel")):
-    verbatim about
+    verbatim markdown(readFile(cfg.staticDir / "md/about.md"))
     h2: text "Instance info"
     p:
       text "Commit "
       a(href=link): text hash
 
-proc renderFeature*(): VNode =
+proc renderFeature*(cfg: Config): VNode =
   buildHtml(tdiv(class="overlay-panel")):
-    verbatim feature
+    verbatim markdown(readFile(cfg.staticDir / "md/feature.md"))
```

Comment:
As far as my little understanding of Nim goes, putting the markdown and readFile calls inside the procedures would result in them running on every request for those routes, which is not the same behaviour as the previous code. Something along these lines below should work better:

```nim
var
  about: string
  feature: string
 
proc renderAbout*(cfg: Config): VNode =
  if (len(about) == 0):
    about = markdown(readFile(cfg.staticDir / "md" / "about.md"))
  buildHtml(tdiv(class="overlay-panel")):
    verbatim about
    h2: text "Instance info"
    p:
      text "Commit "
      a(href=link): text hash

proc renderFeature*(cfg: Config): VNode =
  if (len(feature) == 0):
    feature = markdown(readFile(cfg.staticDir / "md" / "feature.md"))
  buildHtml(tdiv(class="overlay-panel")):
    verbatim feature
```

Reply:
Oh, good catch, thank you!